www/source/partials/docs/_dev-pkgs-share.html.md.erb

```diff
@@ -31,23 +31,23 @@ When you upload a package to Builder, you are required to supply an auth token a
 
 Once you have this token, you can set the `HAB_AUTH_TOKEN` [environment variable](/docs/reference#environment-variables) to this value, so that any commands requiring authentication will use it.
 
-### Create origin keys
+### Create an origin key pair
 
-After you have done the basic account creation steps, you need to create your origin keys. The private key will be used to sign your packages and the public key will be used by supervisors to verify the integrity of your packages (`.hart` files).
+After finishing the basic account creation steps, you need to create your origin key pair. You will use the private key to sign your packages and Supervisors will use the public key to verify the integrity of your packages (`.hart` files).
 
-You can either create an origin key pair by running `hab cli setup` from your host machine, or running `hab origin key generate <ORIGIN>` from either the host machine or from within the studio.
+You can create an origin key pair by running `hab cli setup` from your host machine, or by running `hab origin key generate <ORIGIN>` from either the host machine or from within the studio.
 
-Your keys are located at `~/.hab/cache/keys` on your host machine and `/hab/cache/keys` inside the studio environment.
+Your public and private origin keys are located at `~/.hab/cache/keys` on your host machine and at `/hab/cache/keys` inside the studio environment.
 
 ### Upload the keys for your origin
 
-If you created a new origin and/or Builder does not have the public key that corresponds to the private key used to build your package, you must upload it. You also have the ability to upload your private key; however, if you do not upload at least the public key, Builder will reject the upload of your packages for that origin.
+If you created a new origin and/or Builder does not have the public key corresponding to the private key used to build your package, you must upload it. You also have the ability to upload your private key; however, if you do not upload at least the public key, Builder will reject the upload of your packages for that origin.
```

Comment:
It might be nice to mention why you would *want* to upload your private key to builder